tests/test_utils.py

```diff
@@ -172,3 +174,11 @@ def test_p_values_from_sample():
 
     ks_results = insertion_p_value(ns.insertion[nlive:-nlive], nlive, batch=1)
     assert ks_results['p-value'] > 0.05
+
+
+@pytest.mark.parametrize('method', ['channel', 'kish'])
+def test_effective_samples(method):
+    w = np.ones(20)
+    assert effective_samples(w, method=method) == approx(20, rel=1e-6)
+    w[10:] = 1e-3
+    assert effective_samples(w, method=method) == approx(10, rel=1e-2)
```

Comment:
We should also append the test and check for what behaviour we expect in case there are zeros in `w`, e.g. follow up with the case:
```python
    w[15:] = 0
    ...
```
This would produce warnings and nans in `np.log`, but using scipy's `entropy` resolves that.